odoo/tests/common.py

```diff
@@ -655,11 +655,16 @@ def setUp(self):
         # restore environments after the test to avoid invoking flush() with an
         # invalid environment (inexistent user id) from another test
         envs = self.env.all.envs
+        for env in envs:
+            self.addCleanup(env.clear)
```

Comment:
Iterating over a `WeakSet` is risky, as it may change during the loop.
```suggestion
        for env in list(envs):
            self.addCleanup(env.clear)
```